### PR TITLE
docs: add docs about all cli commands

### LINKED ISSUE #1413: Add README.md files to cli commands
After #1074 is finished we can generate documentation with `neofs-cli gendoc` and move generated files to proper folders.
It would be nice if visiting the package in github displayed the README.md file for the command.

Related #1396.

### PR DESCRIPTION
**Pull request by End-rey**
Generate docs with `neofs-cli gendoc ./docs/cli-commands`.

Closes #1413.

So, if we add a new command, should we generate docs now? Should we also add some checks so that we don't forget to do this?

### CODE REVIEW

**Comment by carpawell** [comment]
> Should we also add some checks so that we don't forget to do this?

I think it will be useful, the same way we check CHANGELOG and configuration files. I am sure I will forget regenerating docs.

File: .github/workflows/cli-docs-update.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            cmd/neofs-cli/**
+
+      - name: Get changed CLI doc files
+        id: docs-diff
+        uses: tj-actions/changed-files@v42
+        with:
+          files: docs/cli-commands/**
+
+      - name: Fail if CLI files are changed but the documentation is not updated
+        if: steps.cli-diff.outputs.any_changed == 'true' && steps.docs-diff.outputs.any_changed == 'false'
```
Comment (roman-khimov) [inline .github/workflows/cli-docs-update.yml:32]: I can change a lot of things in `cmd/neofs-cli` that won't affect documentation in any way. Better regenerate (and maybe even have some `Makefile` target for this) and check for diffs.

File: docs/cli-commands/neofs-cli.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* [neofs-cli control](neofs-cli_control.md)	 - Operations with storage node
+* [neofs-cli gendoc](neofs-cli_gendoc.md)	 - Generate documentation for this command
+* [neofs-cli netmap](neofs-cli_netmap.md)	 - Operations with Network Map
+* [neofs-cli object](neofs-cli_object.md)	 - Operations with Objects
+* [neofs-cli session](neofs-cli_session.md)	 - Operations with session token
+* [neofs-cli storagegroup](neofs-cli_storagegroup.md)	 - Operations with Storage Groups
+* [neofs-cli tree](neofs-cli_tree.md)	 - Operations with the Tree service
+* [neofs-cli util](neofs-cli_util.md)	 - Utility operations
+
+###### Auto generated by spf13/cobra on 28-Oct-2024
```
Comment (roman-khimov) [inline docs/cli-commands/neofs-cli.md:43]: Although this one will trigger every day, can it be removed (at least the date, nothing against `spf13/cobra`, but date is problematic)?

File: pkg/util/gendoc/gendoc.go [hunk 3]
```diff
@@ -14,7 +14,8 @@ import (
 )
 
 const (
-	gendocTypeFlag = "type"
+	gendocDisableAutoGenTagFlag = "disable-auto-gen-tag"
```
Comment (carpawell) [inline pkg/util/gendoc/gendoc.go:17]: not sure we need it, but dont mind, mb @roman-khimov has some thoughts

File: .github/workflows/cli-docs-update.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: cli-docs-check
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Save current CLI docs files
```
Comment (carpawell) [inline .github/workflows/cli-docs-update.yml:18]: can we just try to generate docs and see if `git diff`/\*smth else\* says something new was done?

File: pkg/util/gendoc/gendoc.go (hunk 3 above)
Comment (roman-khimov) [reply pkg/util/gendoc/gendoc.go:17]: I think we do because otherwise we can't have consistent output each time we run the generator.

**Review by roman-khimov — APPROVED** [review]

File: pkg/util/gendoc/gendoc.go (hunk 3 above)
Comment (carpawell) [reply pkg/util/gendoc/gendoc.go:17]: i meant if we need it to be possible to change via separate command flag

File: .github/workflows/cli-docs-update.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Generate new CLI doc files
+        run: make cli-gendoc
+
+      - name: Fail if CLI commands files are changed but the documentation is not updated
+        run: |
+          git add -N docs/cli-commands
+          git diff --exit-code -- docs/cli-commands/
```
Comment (carpawell) [inline .github/workflows/cli-docs-update.yml:24]: can you add new line at the end of file, please?

**Review by carpawell — APPROVED** [review]

**Pull request merged**